library/testing/TestSDKCamera.cxx

```diff
@@ -27,6 +28,31 @@ bool comparePoint(const f3d::point3_t& vec1, const f3d::point3_t& vec2)
     compareDouble(vec1[2], vec2[2]);
 }
 
+class testFailure : public std::runtime_error
+{
+public:
+  explicit testFailure(const std::string& what = "")
+    : std::runtime_error(what)
+  {
+  }
+};
+
+void checkVec3(const std::array<double, 3>& actual, const std::array<double, 3>& expected,
+  const std::string& label)
+{
+  if (!compareDouble(actual[0], expected[0]) || !compareDouble(actual[1], expected[1]) ||
+    !compareDouble(actual[2], expected[2]))
+  {
+    std::stringstream ss;
+    ss << label << ": ";
+    ss << std::setprecision(12);
+    ss << "(" << actual[0] << "," << actual[1] << "," << actual[2] << ")";
+    ss << " != ";
+    ss << "(" << expected[0] << "," << expected[1] << "," << expected[2] << ")";
+    throw testFailure(ss.str());
```

Comment:
any specific reason to go the exceptions instead of conditions (apart from your personnal preference) ?

Reply:
feels like it's less noise to throw on failure and have a single `try {...} catch { std::cerr<< what(); return EXIT_FAILURE; } return EXIT_SUCCESS;` than to have a bunch of `if(!...){ std::cerr<< ... ; return EXIT_FAILURE; }`
...so personal preference mostly I guess